core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -159,6 +159,23 @@ public void log(String format, Object... args) {
         }
     }
 
+    /**
+     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     * @throws RuntimeException which is requested by this call.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
```

Comment:
The reasoning behind this is, we are going to use this to throw unchecked Exception. If we make it Throwable and making it checked exception , it will cause caller to handle it or throw it. 
For example
logger.logAndThrow(new IllegalArgumentException("Null value for poll operation is not allowed."));

Reply:
RuntimeException is fine for now.